src/node.cpp

```diff
@@ -300,7 +300,7 @@ int main(int argc, char * argv[]) {
                     //const int angle_compensate_multiple = 1;
                     const int angle_compensate_nodes_count = 360*angle_compensate_multiple;
                     int angle_compensate_offset = 0;
-                    rplidar_response_measurement_node_hq_t angle_compensate_nodes[angle_compensate_nodes_count];
+                    rplidar_response_measurement_node_hq_t angle_compensate_nodes[angle_compensate_nodes_count+8];
```

Comment:
Why hardcode `8` ? I would understand `1` to add one more time the size of a `rplidar_response_measurement_node_hq_t` to the size of the memory space, but `8` ? This is not a pointer, and even if it was, this an array on the stack, not a malloc-ed buffer, so the size is not counted the same by the language.

If `1` is not the correct option I don't know what is, but `8` is for sure not the right thing to write. The function being huge it is hard to tell what is what but there must be a variable or a formula that makes sense.

Reply:
8 is not the right thing to write. Because I already spent a lot time, I hate this but that's a solution at least. 

Note: I have used Google's AddressSanitizer to detect the corruption. Valgrind or gdb didn't work for me.